src/coreclr/vm/prestub.cpp

```diff
@@ -1499,12 +1498,17 @@ Stub * CreateUnboxingILStubForSharedGenericValueTypeMethods(MethodDesc* pTargetM
     pCode->EmitLoadThis();
     pCode->EmitLDFLDA(tokRawData);
 
-#if defined(TARGET_X86)
+#ifdef TARGET_X86
     // Push the rest of the arguments for x86
     for (unsigned i = 0; i < msig.NumFixedArgs();i++)
     {
         pCode->EmitLDARG(i);
     }
+
+    if (msig.HasAsyncContinuation())
+    {
+        pCode->EmitLDNULL();
```

Comment:
It is correct to pass NULL continuation unconditionally because if we are on resume code path, this code will not run. 

It is a bit subtle, I think. It assumes there is only one suspension point in the stub. 
The stubs have quite a few implicit contracts and requirements, so this is just an observation on one more.

Reply:
It doesn't actually take a dependency on there being only one suspension point in the stub being created here. The only way a function is ever resumed is through its async resumption stub, so the continuation passed by a normal managed call is always null. Same as when the JIT adds the argument -- it is always just a null argument.